Entities/HeldDevice.h

```diff
@@ -281,6 +281,55 @@ ClassInfoGetters
 
 	void SetSupportOffset(Vector newOffset) { m_SupportOffset = newOffset; }
 
+    /// <summary>
+    /// Gets whether this HeldDevice has any limitations on who can pick it up.
+    /// </summary>
+    /// <returns>Whether this HeldDevice has any limitations on who can pick it up.</returns>
+    bool HasPickupLimitations() const { return !m_ActorsWhoCanPickThisUp.empty(); }
+
+    /// <summary>
+    /// Gets whether no Actors (or child classes) can pick up this HeldDevice.
+    /// </summary>
+    /// <returns>Whether no Actors (or child classes) can pick up this HeldDevice.</returns>
+    bool GetNoActorsCanPickThisUp() const { return m_ActorsWhoCanPickThisUp.find(c_NoPickupString) != m_ActorsWhoCanPickThisUp.end(); }
+
+    /// <summary>
+    /// Sets whether no Actors (or child classes) can pick up this HeldDevice.
+    /// </summary>
+    /// <param name="noActors">Whether no Actors (or child classes) should be able to pick up this HeldDevice. True means none will be able to, false means any other limitations will apply normally.</param>
+    void SetNoActorsCanPickThisUp(bool noActors);
+
+    /// <summary>
+    /// Checks whether Actors with the given PresetName can pick up this HeldDevice.
+    /// </summary>
+    /// <param name="actor">The PresetName of the Actor to check. Ownership is NOT transferred.</param>
+    /// <returns>Whether Actors with the given PresetName can pick up this HeldDevice.</returns>
+    bool ActorCanPickThisUp(const std::string &actorPresetName) { return m_ActorsWhoCanPickThisUp.find(actorPresetName) != m_ActorsWhoCanPickThisUp.end(); }
+
+    /// <summary>
+    /// Allow any Actors (or child classes) with the given PresetName to pick up this HeldDevice, as long as it's not set so no Actors can pick it up.
+    /// </summary>
+    /// <param name="actorPresetName">The PresetName of an Actor who should be able to pick up this HeldDevice.</param>
+    void AddActorWhoCanPickThisUp(const std::string &actorPresetName) { m_ActorsWhoCanPickThisUp.insert(actorPresetName); }
```

Comment:
Hmmm, you're totally right about the first part, will change that.

As for the second, adding it back only if it was set in INI is a whole extra variable for this stuff and while it seems reasonable, it also adds inconsistencies that need special handling. I'd say it's better for it to always be consistent, and I can see the argument in favour of making it so when you remove all actors who can pick this up it becomes non pickupable instead of freely pickupable, but I'm not really sure which option is better.